### ISSUE TITLE
SQLITE3 error

### ISSUE THREAD

**Issue by Dirowa** [Avans-ATGM/infrastructure#47]
According to the logs was a few days before the holidays a catasropic update on SQLITE3.
The result of this, is that multiple codes dont work anymore on the galaxy server (bash screen)
The result has for sofar tested that no new data can be uploaded to the galaxy server

the error message:
```shell
Traceback (most recent call last):
  File "metadata/set.py", line 1, in <module>
    from galaxy_ext.metadata.set_metadata import set_metadata; set_metadata()
  File "/srv/galaxy/server/lib/galaxy_ext/metadata/set_metadata.py", line 20, in <module>
    from galaxy.metadata.set_metadata import set_metadata
  File "/srv/galaxy/server/lib/galaxy/metadata/set_metadata.py", line 27, in <module>
    import galaxy.datatypes.registry
  File "/srv/galaxy/server/lib/galaxy/datatypes/registry.py", line 16, in <module>
    from . import (
  File "/srv/galaxy/server/lib/galaxy/datatypes/binary.py", line 27, in <module>
    from galaxy.datatypes.data import (
  File "/srv/galaxy/server/lib/galaxy/datatypes/data.py", line 27, in <module>
    from . import (
  File "/srv/galaxy/server/lib/galaxy/datatypes/dataproviders/__init__.py", line 18, in <module>
    from . import (
  File "/srv/galaxy/server/lib/galaxy/datatypes/dataproviders/dataset.py", line 16, in <module>
    from galaxy.util import sqlite
  File "/srv/galaxy/server/lib/galaxy/util/sqlite.py", line 2, in <module>
    import sqlite3
  File "/usr/lib/python3.8/sqlite3/__init__.py", line 23, in <module>
    from sqlite3.dbapi2 import *
  File "/usr/lib/python3.8/sqlite3/dbapi2.py", line 27, in <module>
    from _sqlite3 import *
ImportError: /usr/lib/python3.8/lib-dynload/_sqlite3.cpython-38-x86_64-linux-gnu.so: undefined symbol: sqlite3_enable_load_extension
```

Luckly more people have this error. I think the best way to proceed to just use ``` apt-get apt --reinstall install sqlite3 ````

or is there something i,m missing (*better safe then sorry when i try something like this) @hexylena

**Comment by Dirowa** [comment]
sofar what tried:
- reinstall sqlite3 no effects
- update apt cache (unavailable due bug)
- reinstall python3 ( no effect)
- install python3.10 (no effect and jenkins doesnt works so reverted back)
- playing with pip

**Comment by hexylena** [comment]
we did the following

- compared the systems `apt list --installed` to list packages, these were rsynced onto the same system, compared with `vimdiff packages*.txt` (other tools work, I just love vim)
- rsynced the dyn-lib directory, and then used `diff -rpnq` to compare the contents, this listed only @Dirowa's test case sqlite .so file
- I made a "minimal working example" with `python -c 'import sqlite3'` which helped us debug
- I ran `ltrace python -c 'import sqlite3'` which traces libraries loaded, didn't have anything that stuck out
- Then I ran `strace python -c 'import sqlite3'` which had something odd, I looked through the logs and saw that splunk's sqlite3 library was loaded after the one we wanted, this really stuck out since it was something DIF added and unexpected to be there
- We looked up how libraries are searched, and found `/etc/ld.so.conf.d/` which had a `splunk.conf` inside. A difference from test-galaxy where it works.
- Disabling that (renaming to splunk.conf.bak and re-running `ldconfig` which updates the search paths) let our MWE work.
